server/src/main/java/org/opensearch/search/query/QueryCollectorContext.java

```diff
@@ -67,38 +68,59 @@
  */
 @PublicApi(since = "1.0.0")
 public abstract class QueryCollectorContext {
-    private static final Collector EMPTY_COLLECTOR = new SimpleCollector() {
-        @Override
-        public void collect(int doc) {}
 
-        @Override
-        public ScoreMode scoreMode() {
-            return ScoreMode.COMPLETE_NO_SCORES;
-        }
-    };
+    private static Collector createEmptyCollector(ScoreMode scoreMode) {
+        return new SimpleCollector() {
+            @Override
+            public void collect(int doc) {}
 
-    public static final QueryCollectorContext EMPTY_CONTEXT = new QueryCollectorContext("empty") {
+            @Override
+            public ScoreMode scoreMode() {
+                return scoreMode;
+            }
+        };
+    }
 
-        @Override
-        Collector create(Collector in) throws IOException {
-            return EMPTY_COLLECTOR;
-        }
+    private static final ReduceableSearchResult EMPTY_RESULT = result -> {};
 
-        @Override
-        CollectorManager<?, ReduceableSearchResult> createManager(CollectorManager<?, ReduceableSearchResult> in) throws IOException {
-            return new CollectorManager<Collector, ReduceableSearchResult>() {
-                @Override
-                public Collector newCollector() throws IOException {
-                    return EMPTY_COLLECTOR;
-                }
+    private static QueryCollectorContext createEmptyContext(String name, Collector collector) {
+        return new QueryCollectorContext(name) {
+            @Override
+            Collector create(Collector in) {
+                return collector;
+            }
 
-                @Override
-                public ReduceableSearchResult reduce(Collection<Collector> collectors) throws IOException {
-                    return result -> {};
-                }
-            };
-        }
-    };
+            @Override
+            CollectorManager<?, ReduceableSearchResult> createManager(CollectorManager<?, ReduceableSearchResult> in) {
+                return new CollectorManager<>() {
+                    @Override
+                    public Collector newCollector() {
+                        return collector;
+                    }
+
+                    @Override
+                    public ReduceableSearchResult reduce(Collection<Collector> collectors) {
+                        return EMPTY_RESULT;
+                    }
+                };
+
+            }
+        };
+    }
+
+    private static String formatContextName(ScoreMode scoreMode) {
+        return String.format(Locale.ROOT, "empty_with_score_mode_%s", scoreMode.toString().toLowerCase(Locale.ROOT));
+    }
+
+    private static final Collector EMPTY_COLLECTOR = createEmptyCollector(ScoreMode.COMPLETE_NO_SCORES);
+    public static final QueryCollectorContext EMPTY_CONTEXT = createEmptyContext(
+        formatContextName(ScoreMode.COMPLETE_NO_SCORES),
```

Comment:
You don't need this `formatContextName`:

```
private static final Collector EMPTY_COLLECTOR = createEmptyCollector(ScoreMode.COMPLETE_NO_SCORES);
private static final Collector EMPTY_COLLECTOR_TOP_SCORES = createEmptyCollector(ScoreMode.TOP_SCORES);

public static final QueryCollectorContext EMPTY_CONTEXT = createEmptyContext(EMPTY_COLLECTOR );
public static final QueryCollectorContext EMPTY_CONTEXT_TOP_SCORES_SCORE_MODE = createEmptyContext(EMPTY_COLLECTOR_TOP_SCORES);
```

The context name of the collector could be deducted from Collector::scoreMode(): `empty_with_score_mode_` + collector.scoreMode().toString().toLowerCase(Locale.ROOT))

Reply:
makes sense, updated the code